### ISSUE TITLE
`=` binding executed twice on initial iteration

### ISSUE THREAD

**Issue by nytpu** [Shinmera/for#9]
Using the latest version from the mainline Quicklisp dist, the `=` binding calls its form twice prior the first iteration, but then functions as normal afterwards. Not an issue when using a functional style because it appears that the first value is ignored, but if the form has side effects then it breaks iteration.

It's also entirely possible I'm using the binding in a way it wasn't intended to be used XD; although if that's the case IMO it should at least be documented that the form shouldn't have side effects. I would rather just use the `=` binding than define a custom iterator since I'm often using it exactly once in a loop where making it have a generalized iterator is wasted effort.

A very contrived example:
``` common-lisp
(for ((ext as 0)
      (i = (incf ext)))
  (format t "~a " i)
  (until (>= i 10)))
```
The above snippet prints `2 3 4 5 6 7 8 9 10 ` instead of `1 2 3 4 5 6 7 8 9 10 ` like one would expect, cf. an equivalent `LOOP` snippet:
``` common-lisp
(loop with ext = 0
      for i = (incf ext)
      do (format t "~a " i)
      until (>= i 10))
```

Looking at the macro expansion, it appears there's a superfluous empty `LET` in the preamble:
``` common-lisp
* (macroexpand-1 '(for ((ext as 0) (i = (incf ext))) (format t "~a " i) (until (>= i 10))))
(WITH-INTERLEAVING
  (WITH-INTERLEAVING
    (LET* ((#:VALUE129 0) (EXT #:VALUE129))
      (DECLARE (IGNORABLE #:VALUE129)))
    NIL)
  (LET ((I (INCF EXT))))  ; <--- superfluous first call
  (LET* ()
    (DECLARE (IGNORABLE)))
  (FOR::WITH-FOR-BLOCK NIL
    (FOR::WITH-FOR-TAGBODY (PROGN
                            (UPDATE I (INCF EXT))  ; <--- main call in body of iteration
                            (FORMAT T "~a " I)
                            (WHEN (>= I 10) (END-FOR)))
      (RETURN-FOR EXT))))
```

**Comment by Shinmera** [comment]
This was already fixed in 0304e7df0fcd6bdac60d9f2926f82cfd67ade34e

**Comment by nytpu** [comment]
Ah, the update must not have made it into Quicklisp yet. I'll pull in the latest version directly from here then, thanks!
